### PR TITLE
feat: Add a test for supported types of SortMergeJoin in DataFusion

### LINKED ISSUE #357: Add a test for supported types of SortMergeJoin in DataFusion
### What is the problem the feature request solves?

From [planga82](https://github.com/planga82)'s comment: https://github.com/apache/datafusion-comet/pull/355#issuecomment-2086056132

> As an idea, I was thinking that we could put in tests to check the supported types in Datafusion, so that when we upgrade to a version of datafusion that supports these other types (like timestamp with timezone for example) we are aware that we should change the list of allowed types.

### Describe the potential solution

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by planga82**
## Which issue does this PR close?

Closes #357 .

## Rationale for this change

The goal of these tests is that if Datafusion supports more data types, we remember to support them here.

## What changes are included in this PR?

Add new tests

## How are these changes tested?

Unit testing
We have yet configuration to support testing (COMET_EXEC_BROADCAST_FORCE_ENABLED) so this PR add new configuration to support the tests.
I can't do complex type testing because we don't support it in most operators.

### CODE REVIEW

File: common/src/main/scala/org/apache/comet/CometConf.scala [hunk 1]
```diff
@@ -383,6 +383,13 @@ object CometConf {
     .booleanConf
     .createWithDefault(false)
 
+  val COMET_SORTMERGEJOIN_CHECK_KEY_TYPES: ConfigEntry[Boolean] =
```
Comment (comphead) [inline common/src/main/scala/org/apache/comet/CometConf.scala:386]: thanks @planga82 
I'm not sure if we need to change the conf for the unit test purpose

Comment (planga82) [reply common/src/main/scala/org/apache/comet/CometConf.scala:386]: thanks @comphead . Any idea how to do this in a cleaner way? We could move this constant to other place and avoid putting it in CometConf. In that way we avoid to expose a testing configuration. What do you think?

Comment (comphead) [reply common/src/main/scala/org/apache/comet/CometConf.scala:386]: Tbh to test variety of supported datatypes I liked the @andygrove approach introduced in https://github.com/apache/datafusion-comet/pull/351, wondering if we can do that

Comment (planga82) [reply common/src/main/scala/org/apache/comet/CometConf.scala:386]: I'm going to review it. Thanks!

Comment (planga82) [reply common/src/main/scala/org/apache/comet/CometConf.scala:386]: I think it's not the same idea and not applicable here. I'll close this PR and try to think of another way to do it without adding configuration. Thanks again!

**Pull request closed without merging**